client/src/components/Loading/Loading.js

```diff
@@ -1,62 +1,89 @@
-/* global window */
-import React, { PureComponent } from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { v4 as uuid } from 'uuid';
 
 /**
  * Renders a Silverstripe loading animation styled after the Silverstripe logo
+ *
+ * `spinner__animation__outline_right` and `spinner__animation__outline_left`
+ * They form the two halves of the logo. They are transparent atd are use
+ * as masks to clip the things that are animated.
+ *
+ * spinner__animation__fill-left and spinner__animation__fill-right
+ * Those are two plain lines with heavy strokes. The lines mostly follow the
+ * middle of the two shapes. The strokes are dashed with a dash length long
+ * enough to cover the full length of the two paths. SVG allow you to control
+ * the dash offset. That's what's actually animated to create the illusion
+ * that the paths are being "drawn".
  */
-class Loading extends PureComponent {
-  render() {
-    const { containerClass } = this.props;
+function Loading({ containerClass }) {
+  // If we are displaying many loading indicators at the same time, we need to
+  const [id] = useState(uuid());
 
-    return (
-      <div className={containerClass}>
-        <div key="overlay" className="cms-content-loading-overlay ui-widget-overlay-light" />
-        <div key="spinner" className="cms-content-loading-spinner">
-          <div className="spinner">
-            {/* window.location is neccessary for Safari which otherwise prepends the base tag */}
```

Comment:
I've decided to throw Safari under the bus here. Basically, forcing us to use absolute URL to find those SVGs introduce a bug where sometime if you've updated the URL with react route or added some extra parameter, the SVG path clipping doesn't work.

Using the IDs directly instead of the full URL seems to mitigate this problem.

Basically, I think that having this work consistently in all other browser is better than adding a hack so that Safari users get it looking at it's best some of the time.

Safari user should still get a reasonable animation in most cases. It just won't have the nice curvy bit. Given that those loading animation are only shown for a fleeting second and are pretty small, I doubt anyone will even notice.

Reply:
Safari will probably look something like this.

![image](https://github.com/silverstripe/silverstripe-admin/assets/1168676/87622906-36f7-4960-b889-beadafe43091)